### PR TITLE
Update of coq packages wrt ocaml versions (8.2, 8.3, 8.4)

### PR DESCRIPTION
**Pull request by herbelin**
Split out of #240 for only 8.2, 8.3 and 8.4. See discussion there.

### CODE REVIEW

**Comment by mattam82** [comment]
A patch seems to be needed to make 8.2 compile with recent OCaml compilers. Or an upper bound on the compiler version.

**Comment by herbelin** [comment]
> A patch seems to be needed to make 8.2 compile with recent OCaml compilers. Or an upper bound on the compiler version.

Indeed, my changed were still in PR state and not pushed. Now pushed, trying again...

**Comment by mattam82** [comment]
There seems to be an issue with lablgtk installation, didn't it work for the 8.5/8.6 packages?

**Comment by RalfJung** [comment]
I fixed the GTK stuff on Linux, and CI passes there. I don't know Apple enough to fix CI there. I'd be fine with merging anyway...

For 8.5/8.6 packages, CoqIDE wasn't built.

File: core-dev/packages/coq.8.2.dev/opam [hunk 1]
```diff
@@ -2,6 +2,7 @@ opam-version: "1.1"
 maintainer: "hugo.herbelin@inria.fr"
```
Comment (gares) [inline core-dev/packages/coq.8.2.dev/opam]: The `opam lint` command complains here, it wants 1.2 (as was done for other versions of the package)

**Comment by mattam82** [comment]
@herbelin @RalfJung ping :) It seems with the minor fix @gares suggested we could merge this. I can try the OS X build if you like

**Comment by RalfJung** [comment]
I can't really do anything here. Mac CI fails, we can ignore it or find someone who knows how to compile CoqIDE on Mac.

**Comment by mattam82** [comment]
I do, I mean I did :)
Le mer. 14 févr. 2018 à 14:04, Ralf Jung <notifications@github.com> a
écrit :

**Comment by mattam82** [comment]
Let me try and report then

**Comment by herbelin** [comment]
Oups, I had prepared an answer but pushed the wrong buttong and it remained pending. The questions I had were: what are the implications of requiring opam 1.1 or 1.2? Why can't we be more general by accepting both 1.1 and 1.2? What are the constraints for the users to require 1.2? How to interpret the message "error 21: Field 'opam-version' doesn't match the current version, validation may not be accurate"?

**Comment by mattam82** [comment]
As the opam linter is 1.2 it needs 1.2 versions. I think the opam files themselves are 1.1-compatible, but anyway the 1.1 version of opam will fail in other respects. I think most everyone is on the 1.2 format these days. The error 21 is probably about trying to lint an 1.1 formated file with an opam using the 1.2 format

**Comment by herbelin** [comment]
So, do you mean that we shall consistently adopt the rule to expect opam 1.2 for all users (and all packages)? [At first view, there are 8 other packages like this to change then.]

**Comment by mattam82** [comment]
I think so yes. @gares right?

**Comment by gares** [comment]
Yes

**Comment by herbelin** [comment]
Well, I moved the opam requirement from 1.1 to 1.2 for 8.2. See also #267 for the other packages.

**Comment by RalfJung** [comment]
@mattam82 MacOS GTK stuff still failing:
```
### stderr ###
# Package gtk+-2.0 was not found in the pkg-config search path.
# Perhaps you should add the directory containing `gtk+-2.0.pc'
# to the PKG_CONFIG_PATH environment variable
# No package 'gtk+-2.0' found
# configure: error: GTK+ is required
[...]
The following actions failed
  ∗  install lablgtk 2.18.5
```

**Comment by mattam82** [comment]
Ok. Let me try now.

**Comment by mattam82** [comment]
Works here with 8.2.dev, I get a working coqide.

**Comment by mattam82** [comment]
There's a little issue with removing 8.2.dev (and I guess other packages). When I updated to coq.8.3.dev, I couldn't launch coqide because the initial.coq file was presumably not removed correctly. Shall we add `remove` targets to the opam files to fix this?

**Comment by herbelin** [comment]
> Shall we add remove targets to the opam files to fix this?

Looks like a good idea. What is the general methodology to build such `remove` target?

**Comment by RalfJung** [comment]
> Looks like a good idea. What is the general methodology to build such remove target?

I'd love to know that as well. :D  The `remove` target of the latest version(s) is written by manually checking that it removes all the files that got installed...

**Comment by mattam82** [comment]
Uh-oh. Maybe I was too optimistic, it sounds like a can of worms :)

**Pull request merged**